odxtools/diaglayer.py

```diff
@@ -256,60 +338,141 @@ def diag_data_dictionary_spec(self) -> DiagDataDictionarySpec:
     #####
     # <value inheritance mechanism helpers>
     #####
-    def _get_parent_refs_sorted_by_priority(self, reverse=False):
+    def _get_parent_refs_sorted_by_priority(self, reverse=False) -> Iterable[ParentRef]:
         return sorted(
             self.diag_layer_raw.parent_refs,
             key=lambda pr: pr.layer.variant_type.inheritance_priority,
             reverse=reverse)
 
-    def __gather_local_services(
-            self, odxlinks: OdxLinkDatabase) -> List[Union[DiagService, SingleEcuJob]]:
-        diagcomms_by_name: Dict[str, Union[DiagService, SingleEcuJob]] = {}
+    def _compute_available_objects(
```

Comment:
_compute_available_objects not used to simplify protocols and _compute_available_commmunication_parameters ?

Reply:
protocols are not inherited, but in some sense they *are* the inheritance hierarchy. Regarding the comparams, for some arcane reason, they use a slightly different inheritance mechanism: https://github.com/mercedes-benz/odxtools/pull/146/files#diff-3a939bdfc46d96678ce4207d04602c4123b800b483f627e8288cab31d3c2270bR352 (cf section 7.3.6). It's very cool that you notice such things, I probably wouldn't ;)

Reply:
Could you put this as comments? so that the next guy does not try to "simplify" the code and break logic?

Reply:
Other than this, LGTM

Reply:
good point. done: 84003041aa08c2c223c532a7b2aac2f2c1b737de